packages/embark-specialconfigs/src/index.js

```diff
@@ -273,4 +276,16 @@ class SpecialConfigs {
   }
 }
 
+function createDeployHookWithCustomLogContext(logPrefix, deployHook) {
+  return async function (dependencies) {
+    const originalLogFn = console.log;
+    console.log = function(...args) {
+      args.forEach(arg => originalLogFn(`${logPrefix} ${arg}`));
+    }
+    const result = await deployHook(dependencies);
+    console.log = originalLogFn;
+    return result;
+  }
+}
```

Comment:
Instead of monkey-patching `console.log` temporarily we can also take advantage of Embark's `Logger` and use its `context` to mutate the original log output. This however, would require users to use `logger` instead of `console.log`, which is less "intuitive". E.g:

```
onDeploy: (context) => {
  context.logger.info('Hello from onDeploy');
}
```

We would then take care of setting logger's context accordingly. However, as mentioned, ppl that won't use logger, will still end up with non-prefixed output.

OTOH, overwriting `console.log` feels a little dirty too, although it's only temporarily for the moment the deployment hook is called.

Reply:
I think overriding console.log is fine, because like you said, no one uses or would use embark's logger.